src/main/java/sirius/biz/storage/layer3/DeleteFilesJob.java

```diff
@@ -45,7 +50,81 @@ protected DeleteFilesJob(ProcessContext process) {
 
     @Override
     public void execute() throws Exception {
+        VirtualFile sourcePath = process.require(SOURCE_PATH_PARAMETER);
+        if (!sourcePath.exists() || !sourcePath.isDirectory()) {
+            process.log(ProcessLog.error()
+                                  .withNLSKey("DeleteFilesJob.sourcePath.invalid")
+                                  .withContext(PATH_KEY, sourcePath.path()));
+            return;
+        }
+        handleDirectory(sourcePath);
+    }
+
+    private boolean handleDirectory(VirtualFile directory) {
```

Comment:
**Optional:** Would like some small JavaDoc on these methods to explain the rough logic / flow